django/db/models/sql/where.py

```diff
@@ -118,6 +118,17 @@ def as_sql(self, compiler, connection):
         None, [] if this node is empty, and raise EmptyResultSet if this
         node can't match anything.
         """
+        if self.negated and all(
+            isinstance(child, WhereNode) and child.negated for child in self.children
+        ):
+            return self.create(
+                [
+                    child.create(child.children, child.connector)
+                    for child in self.children
+                ],
+                self.connector,
```

Comment:
```suggestion
                AND if self.connector == OR else AND,
```

Actually yep I think we need to flip this.  Also we can't branch here if `self.connector` is XOR (there is no de morgan for xor is there? 🤔

Reply:
Also couple of more things:

- Would be good to add a test specifically for this mini-featurette
- A comment might be nice for future maintainers that don't recognise the pattern that we're applying de morgan

Reply:
You are completely right @shangxiao, I'll make sure to add tests for this particular change and make the propose adjustments.